src/main/webapp/app/course/tutorial-groups/tutorial-groups-management/tutorial-free-periods/crud/tutorial-free-period-form/tutorial-group-free-period-form.component.ts

```diff
@@ -1,44 +1,150 @@
 import { faCalendarAlt } from '@fortawesome/free-solid-svg-icons';
-import { TutorialGroupSessionFormData } from 'app/course/tutorial-groups/tutorial-groups-management/tutorial-group-sessions/crud/tutorial-group-session-form/tutorial-group-session-form.component';
+import { TutorialGroupFreePeriodsManagementComponent } from 'app/course/tutorial-groups/tutorial-groups-management/tutorial-free-periods/tutorial-free-periods-management/tutorial-group-free-periods-management.component';
+import { TutorialGroupFreePeriod } from 'app/entities/tutorial-group/tutorial-group-free-day.model';
+import dayjs from 'dayjs/esm';
 
 export interface TutorialGroupFreePeriodFormData {
-    date?: Date;
+    startDate?: Date;
+    endDate?: Date;
+    startTime?: Date;
+    endTime?: Date;
     reason?: string;
 }
+
+// ToDo: TimeFrame Enum
+export enum TimeFrame {
+    Day,
+    Period,
+    PeriodWithinDay,
+}
+
 @Component({
     selector: 'jhi-tutorial-free-period-form',
     templateUrl: './tutorial-group-free-period-form.component.html',
     changeDetection: ChangeDetectionStrategy.OnPush,
 })
+
+// ToDo: extend TutorialGroupFreePeriodFormData to support an endDate
 export class TutorialGroupFreePeriodFormComponent implements OnInit, OnChanges {
     @Input()
     formData: TutorialGroupFreePeriodFormData = {
-        date: undefined,
+        startDate: undefined,
+        endDate: undefined,
+        startTime: undefined,
+        endTime: undefined,
         reason: undefined,
     };
 
     @Input() isEditMode = false;
 
     @Input() timeZone: string;
 
-    @Output() formSubmitted: EventEmitter<TutorialGroupSessionFormData> = new EventEmitter<TutorialGroupSessionFormData>();
+    @Output() formSubmitted: EventEmitter<TutorialGroupFreePeriodFormData> = new EventEmitter<TutorialGroupFreePeriodFormData>();
 
     faCalendarAlt = faCalendarAlt;
 
     form: FormGroup;
 
-    get dateControl() {
-        return this.form.get('date');
+    protected timeFrame = TimeFrame.Day;
+
+    protected readonly TimeFrame = TimeFrame;
+
+    // Todo: TimeFrame getter/setter. Reset endDate when switching back to single Day
+    setTimeFrame(timeFrame: TimeFrame) {
+        if (timeFrame === TimeFrame.Day) {
+            this.resetDateControl('endDate');
+            this.resetDateControl('endTime');
+            this.resetDateControl('startTime');
+        } else if (timeFrame == TimeFrame.Period) {
+            this.resetDateControl('startTime');
+            this.resetDateControl('endTime');
+        } else if (timeFrame == TimeFrame.PeriodWithinDay) {
+            this.resetDateControl('endDate');
+        }
+        this.timeFrame = timeFrame;
+    }
+
+    private resetDateControl(controlName: string) {
+        if (this.form.get(controlName)) {
+            this.form.get(controlName)?.reset();
+            this.form.get(controlName)?.markAsUntouched();
+        }
+    }
+
+    get isStartBeforeEnd() {
```

Comment:
Add this functionality also to the server side.

Reply:
done